sdk/keyvault/azure-keyvault-keys/tests/_test_case.py

```diff
@@ -29,23 +30,30 @@ def wrapper(test_class_instance, api_version, is_hsm=False, **kwargs):
             loop.run_until_complete(coroutine)
         else:
             testcase_func(test_class_instance, client, is_hsm=is_hsm)
+
     return wrapper
 
 
-def get_decorator(hsm_only=False, vault_only=False, **kwargs):
+def get_decorator(only_hsm=False, only_vault=False, **kwargs):
     """returns a test decorator for test parameterization"""
-    params = [param(api_version=p[0], is_hsm=p[1], **kwargs) for p in get_test_parameters(hsm_only, vault_only)]
+    versions = kwargs.pop("api_versions", None) or ApiVersion
```

Comment:
This would be okay as a parameter because there are no compatibility concerns or design guidelines affecting test code